libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -359,13 +365,18 @@ void AP_Frsky_Telem::send_SPort(void)
 void AP_Frsky_Telem::send_D(void)
 {
     const AP_BattMonitor &_battery = AP::battery();
+    int8_t battery_pct;
+    if(!_battery.capacity_remaining_pct(battery_pct)) {
+        battery_pct = -1;
+    }
+
     uint32_t now = AP_HAL::millis();
     // send frame1 every 200ms
     if (now - _D.last_200ms_frame >= 200) {
         _D.last_200ms_frame = now;
         send_uint16(DATA_ID_TEMP2, (uint16_t)(AP::gps().num_sats() * 10 + AP::gps().status())); // send GPS status and number of satellites as num_sats*10 + status (to fit into a uint8_t)
         send_uint16(DATA_ID_TEMP1, gcs().custom_mode()); // send flight mode
-        send_uint16(DATA_ID_FUEL, (uint16_t)roundf(_battery.capacity_remaining_pct())); // send battery remaining
+        send_uint16(DATA_ID_FUEL, (uint16_t)roundf(battery_pct)); // send battery remaining
```

Comment:
Pre-exsisting, no need to change it, but we really should come back and remove the roundf call, it's just wasting time.